frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx

```diff
@@ -306,10 +316,29 @@ export function ObjectDetailView({
         actions,
         databases,
         onDelete: () => "TODO: metabase#32323",
-        onUpdate: () => "TODO: metabase#32322",
+        onUpdate: action => setExecuteActionId(action.id),
       })
     : [];
 
+  const fetchInitialValues = useCallback(async () => {
+    if (typeof executeActionId !== "number") {
+      return {};
+    }
+
+    return ActionsApi.prefetchValues({
+      id: executeActionId,
+      parameters: JSON.stringify({ id: String(zoomedRowID) }),
+    });
+  }, [executeActionId, zoomedRowID]);
+
+  const initialValues = useMemo(() => ({ id: zoomedRowID }), [zoomedRowID]);
+
+  const dispatch = useDispatch();
+
+  const handleActionSuccess = useCallback(() => {
```

Comment:
do you really need `useCallback` for modal actions?

Reply:
:+1: 357d720eea143b4263c84f7d90a114a05536202b